backend/collect/graphs.py

```diff
@@ -0,0 +1,44 @@
+from typing import List
+from rdflib import Graph, RDF, IdentifiedNode
+from rdflib.term import Node
+
+from triplestore.utils import Triples
+
+def list_from_graph_collection(graph: Graph, list_node: IdentifiedNode) -> List[Node]:
+    '''
+    Extract a list of nodes from an RDF collection in a graph
+    '''
+
+    items = list(graph.objects(list_node, RDF.first))
+    rest_nodes = graph.objects(list_node, RDF.rest)
+    for rest in rest_nodes:
+        items += list_from_graph_collection(graph, rest)
+    return items
+
+
+def list_to_graph_collection(items: List[Node], items_node: IdentifiedNode) -> Graph:
+    '''
+    Return a list of items as an RDF collection
+    '''
+    
+    g = Graph()
+    collection = g.collection(items_node)
+    collection += items
+    return g
+
+
+def collection_triples(graph: Graph, list_node: IdentifiedNode) -> Triples:
```

Comment:
The rationale here was that the graph may be implemented in a non-standard way, such as having multiple `first` or `rest` elements for a list node. If we're deleting or changing a collection, it's relevant to select the triples that are currently stored, rather than reconstruct how they *should* be stored.

You could compare these results to see the difference:

```python
found_triples = collection_triples(g, node)

items = list_from_graph_collection(g, node)
reconstructed_collection = list_to_graph_collection(items, node)
normalized_triples = all_triples(reconstructed_collection)
```

The second is essentially what you're proposing, which will yield the same result if the list is stored the way you would expect. (If you use `model.save()`, it will replace the `found_triples` with the `normalized_triples`, by the way.)

So in theory yes, these would be the same, but they might not be and these functions have different uses in that scenario. I'll add a comment to explain this.